operations/mimir/servicemonitor.libsonnet

```diff
@@ -0,0 +1,51 @@
+{
+  local this = self,
+
+  serviceMonitor: if $._config.create_service_monitor then {
+    apiVersion: 'monitoring.coreos.com/v1',
+    kind: 'ServiceMonitor',
+    metadata: {
+      name: 'mimir',
+      namespace: $._config.namespace,
+    },
+    spec: {
+      namespaceSelector: {
+        matchNames: [$._config.namespace],
+      },
+      servicesToRelabel:: [
```

Comment:
Do we also want to add the
* ruler
* the ruler-(query-frontend|querier|query-scheduler) 
* and overrides-exporter components?

Reply:
this PR with one servicemontor selecting many services is inspired from [loki](https://github.com/grafana/loki/blob/main/production/ksonnet/loki/servicemonitor.libsonnet), 

another way to do it is how it's done in the chart: one servicemonitor per service to monitor: https://github.com/search?q=repo%3Agrafana%2Fmimir%20mimir.lib.serviceMonitor&type=code, then there is no need for this "centralized"